projectq/cengines/_basics.py

```diff
@@ -167,16 +181,14 @@ def is_meta_tag_supported(self, meta_tag):
             returns True.
         """
         engine = self
-        try:
-            while True:
-                try:
-                    if engine.is_meta_tag_handler(meta_tag):
-                        return True
-                except AttributeError:
-                    pass
-                engine = engine.next_engine
-        except:
-            return False
+        while engine is not None:
+            try:
+                if engine.is_meta_tag_handler(meta_tag):
+                    return True
+            except AttributeError:
+                pass
+            engine = engine.next_engine
+        return False
```

Comment:
Getting rid of the first `try` statement can be safely done since I removed the infinite loop.
For the second one, the possible expected error is an `AttributeError` if the engine does not have the `is_meta_tag_handler` method (which it by default does not have).

Reply:
Yes, I understood that you removed the infinite loop.
I just meant that if those two operation raise an exception different from `AttributeError` for whatever reason, the old function returned just `False`, while the new version propagates (i.e., does not catch) the exception.

Reply:
Yes this is intended. Any other exception should not be silently ignored as it is an error that is not expected by ProjectQ itself.

Reply:
That's what I thought. Thanks for confirming.